openeogeotrellis/job_tracker_v2.py

```diff
@@ -85,48 +87,112 @@ class YarnAppReportParseException(Exception):
+            )
 
-    def parse_application_report(self, report: str) -> _JobMetadata:
-        try:
-            props = dict(re.findall(r"^\t(.+?) : (.+)$", report, flags=re.MULTILINE))
+        report = json.get("app", {})
+        required_keys = ["state", "finalStatus", "startedTime", "finishedTime"]
+        missing_keys = [k for k in required_keys if k not in report]
+        if missing_keys:
+            raise YarnAppReportParseException(
+                f"JSON response is missing following required keys: {missing_keys}"
+            )
 
+        try:
             job_status = yarn_state_to_openeo_job_status(
-                state=props["State"], final_state=props["Final-State"]
+                state=report["state"], final_state=report["finalStatus"]
             )
+            # Log the diagnostics if there was an error.
+            # In particular, sometimes YARN can't launch the container and then the
+            # field 'diagnostics' provides more info why this failed.
+            diagnostics = report.get("diagnostics", None)
+            if job_status == JOB_STATUS.ERROR and diagnostics:
+                _log.error(
+                    f"YARN application status reports error diagnostics: diagnostics: {diagnostics}"
+                )
 
-            def ms_epoch_to_date(epoch_millis: str) -> Union[str, None]:
-                """Parse millisecond timestamp from app report and return as rfc3339 date (or None)"""
-                if epoch_millis == "0":
-                    return None
-                utc_datetime = dt.datetime.utcfromtimestamp(int(epoch_millis) / 1000)
-                return rfc3339.datetime(utc_datetime)
-
-            start_time = ms_epoch_to_date(props["Start-Time"])
-            finish_time = ms_epoch_to_date(props["Finish-Time"])
-
-            allocation = props["Aggregate Resource Allocation"]
-            match = re.fullmatch(r"^(\d+) MB-seconds, (\d+) vcore-seconds$", allocation)
-            if match:
-                usage = {
-                    "cpu": {"value": int(match.group(2)), "unit": "cpu-seconds"},
-                    "memory": {"value": int(match.group(1)), "unit": "mb-seconds"},
-                }
-            else:
+            start_time = cls._ms_epoch_to_date(report["startedTime"])
+            finish_time = cls._ms_epoch_to_date(report["finishedTime"])
+
+            if not ("memorySeconds" in report or "vcoreSeconds" in report):
                 usage = None
+            else:
+                usage = {}
+                memory_seconds = report.get("memorySeconds")
+                vcore_seconds = report.get("vcoreSeconds")
+                if memory_seconds is not None:
+                    usage["memory"] = {
+                        "value": int(memory_seconds),
+                        "unit": "mb-seconds",
+                    }
+                if vcore_seconds is not None:
+                    usage["cpu"] = {"value": int(vcore_seconds), "unit": "cpu-seconds"}
```

Comment:
Not exactly the behavior that we had before, but I would just try to log memory seconds and vcore seconds if they're available, regardless of whether or not one is missing.

Reply:
Good idea, much simpler code. 
Done.